### PR TITLE
SJ INTERFACE ENABLES RESUMING JOBS: As Ting, I want to be able to resume my Emu harvest / enrichment, so that my marathon harvest can be completed even with surprise stoppages

### PR DESCRIPTION
**Pull request by paul-mesnilgrente**
[SJ INTERFACE ENABLES RESUMING JOBS: As Ting, I want to be able to resume my Emu harvest / enrichment, so that my marathon harvest can be completed even with surprise stoppages](https://www.pivotaltracker.com/story/show/175649486)

STORY
=====

**Acceptance Criteria**
- All these [Stopped jobs](https://manager.digitalnz.org/production/jobs?status=stopped) do not show a `Stop Enrichment` button when viewing their `Details` page.
- Button should only show for relevant jobs (json paginated parsers or xml files)
- [Job details](https://manager.digitalnz.org/production/enrichment_jobs/5fb241764e617a0001059fda) pages that have a status of 'stopped' or 'failed' show a `Resume Enrichment` or `Resume Harvest` button where the 'Stop ...' button sometimes is, see attachment.
- Pressing the button shows confirmation message
>This will attempt to resume this job from its previous state.
> Are you sure? YES / NO
- Pressing NO returns to job details page
- Pressing YES resumes the harvest / enrichment 
- Resumed jobs keep a message (maybe in [Status Message](https://manager.digitalnz.org/production/enrichment_jobs/5fb241764e617a0001059fda) ) saying "Harvest was resumed November 18, 2020 09:42:10 

**Risks**
- Resuming the wrong(?) job corrupts the metadata somehow

**Notes**
- It is assumed the resumed job will continue updating the various information on the `Details` page
- It is expected this will be tested in UAT however the SJ UAT pipeline needs to be fixed prior to this.

SCREENSHOTS
============

### CODE REVIEW

File: spec/factories/harvest_job.rb [hunk 1]
```diff
@@ -9,5 +9,35 @@
...
+
+    trait :finished do
+      id { 1 }
+      status { 'finished' }
+    end
+
+    trait :failed do
+      id { 1 }
+      status { 'failed' }
+    end
```
Comment (invke) [inline spec/factories/harvest_job.rb:41]: The `id { 1 }` makes these traits not very useable outside of specific scenarios where no other HarvestJob's exist. If you could remove it and decouple the factory from the test that'd be a win imo :)

**Review by invke — APPROVED** [review]
Looks all good man, I think I am following what's going on 🤷‍♂️ 

Just a couple of miscellaneous thoughts.

File: app/models/concerns/template_helpers.rb [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+
+  def update_contents_parser_class!
+    class_name = name.gsub(/\s/, '_').camelize
+    self.content = content.gsub(/^class .* </, "class #{class_name} <")
+    self.message = 'Renamed parser class'
+    self.save
+  end
+
+  def apply_parser_template!
+    if self.content.nil?
```
[the remark this answers is not in the document]
Comment (paul-mesnilgrente) [reply app/models/concerns/template_helpers.rb:14]: Ah I actually didn't create that method, I moved it, will do anyway I think

File: app/models/concerns/template_helpers.rb [hunk 3]
```diff
@@ -0,0 +1,30 @@
+# frozen_string_literal: true
+
+module TemplateHelpers
+  extend ActiveSupport::Concern
+
+  def update_contents_parser_class!
+    class_name = name.gsub(/\s/, '_').camelize
```
[the remark this answers is not in the document]
Comment (paul-mesnilgrente) [reply app/models/concerns/template_helpers.rb:7]: The names have spaces in them, it's basically a parser name like: 'Harvesting the national library' that we translate to be a class name for the parser script: 'HarvestingTheNationalLibrary'

File: app/views/enrichment_jobs/_enrichment_job.html.erb [hunk 4]
```diff
@@ -1,4 +1,16 @@
-<%= content_tag :div, id: "enrichment-job", data: { url: environment_enrichment_job_path(params[:environment], @enrichment_job, format: "js"), active: !@enrichment_job.finished? } do %>
+<%=
+  tag.dig(
```
[the remark this answers is not in the document]
Comment (paul-mesnilgrente) [reply app/views/enrichment_jobs/_enrichment_job.html.erb:2]: yup!

File: app/views/enrichment_jobs/_enrichment_job.html.erb [hunk 5]
```diff
@@ -8,22 +20,48 @@
...
+          environment: params[:environment],
+          id: @enrichment_job.id,
+          format: :js,
+          enrichment_job: { status: "stopped" }
+        ),
+        method: :put,
+        remote: true,
+        id: "stop-enrichment-button",
+        class: "button new-right"
+      )
```
[the remark this answers is not in the document]
Comment (paul-mesnilgrente) [reply app/views/enrichment_jobs/_enrichment_job.html.erb:38]: Yeah true, will update

File: app/models/concerns/template_helpers.rb (hunk 2 above)
[the remark this answers is not in the document]
Comment (paul-mesnilgrente) [reply app/models/concerns/template_helpers.rb:14]: All done

File: app/models/concerns/template_helpers.rb (hunk 3 above)
[the remark this answers is not in the document]
Comment (paul-mesnilgrente) [reply app/models/concerns/template_helpers.rb:7]: Extracted to a protected method for clarity

File: app/views/enrichment_jobs/_enrichment_job.html.erb (hunk 4 above)
[the remark this answers is not in the document]
Comment (paul-mesnilgrente) [reply app/views/enrichment_jobs/_enrichment_job.html.erb:2]: All fixed

**Review by motizuki — APPROVED** [review]
Good job guys!

File: app/views/enrichment_jobs/_enrichment_job.html.erb [hunk 6]
```diff
@@ -9,21 +21,50 @@
...
-  %>
+  <% if @enrichment_job.status == 'active' %>
+    <%=
+      link_to(
+        t("enrichment_jobs.stop"),
+        environment_enrichment_job_path(
+          environment: params[:environment],
+          id: @enrichment_job.id,
+          format: :js,
+          enrichment_job: { status: "stopped" }
```
Comment (EdwinRozario) [inline app/views/enrichment_jobs/_enrichment_job.html.erb:32]: Use single quotes

**Review by EdwinRozario — APPROVED** [review]

**Pull request merged**